src/module/migration.ts

```diff
@@ -7,29 +9,47 @@ import {TwodsixItemData, UpdateData} from "../types/twodsix";
  */
 export class Migration {
 
-  private static async migrateActorData(actor:Actor):Promise<UpdateData> {
+  private static async migrateActorData(actor:TwodsixActor):Promise<UpdateData> {
     const updateData:UpdateData = <UpdateData>{};
-    const actorData = actor.data;
-    await this.migrateActorItems(actorData, actor);
+
+    let untrainedSkill = actor.getUntrainedSkill();
+    if (!untrainedSkill) {
+      untrainedSkill = await actor.buildUntrainedSkill();
+      updateData['data.untrainedSkill'] = untrainedSkill._id;
+    }
+
+    //TODO Get rid of the untrainedSkill passing
+    await this.migrateActorItems(actor, untrainedSkill);
 
     return updateData;
   }
 
-  private static async migrateActorItems(actorData:ActorData, actor:Actor) {
+  private static async migrateActorItems(actor:TwodsixActor, untrainedSkill:TwodsixItem=null) {
     //Handle any items that are on the actor
-    const actorItems = actorData["items"];
+    const actorItems = actor.data["items"];
     const toUpdate = [];
     for (const i of actorItems) {
-      toUpdate.push(mergeObject(i, this.migrateItemData(i)));
+      toUpdate.push(mergeObject(i, this.migrateItemData(i, actor, untrainedSkill)));
     }
     await actor.updateEmbeddedEntity("OwnedItem", toUpdate);
   }
 
-  private static migrateItemData(item:TwodsixItemData):UpdateData {
+  private static migrateItemData(item:TwodsixItemData, actor:TwodsixActor = null, untrainedSkill:TwodsixItem=null):UpdateData {
     const updateData:UpdateData = <UpdateData>{};
 
-    if (item.type === 'skills') {
+    if (item.type === 'skills') { //0.6.82
       updateData['data.rolltype'] = item.data.rolltype || 'Normal';
+      if (typeof item.data.value ===  "string") { // 0.7.8
```

Comment:
Perhaps we should use a system with versioned migrations? I'm thinking that each migration would be named after a date and time, and it would only run a migration once and remember which migration you have run. Now we might get bugs down the line if we don't remember how the system looked like in version 0.6.5 for example.

Reply:
Heh. I just left that system as I thought that was bad. :) Not saying I was right about that though. This just seemed easier to not get wrong. As in, no matter the reason that you don't have that field, now you get it.
Anyway, check previous versions to see how it was done until quite recently.
And, to get rid of all the historical migrations that don't matter anymore.

Reply:
Ah, I see now how it was done. I imagined something  slightly different! What I meant was that each file would be a separate independent script, an example of a recent migration would then  be:

2021_01_10_21_55_51231231_migration.js:
```
console.log("Starting migration: 2021_01_10_21_55_51231231");
try {
  game.actors.collection.forEach(async actor => {
    let untrainedSkill = actor.getUntrainedSkill();
    if (!untrainedSkill) {
      untrainedSkill = await actor.buildUntrainedSkill();
      console.log(`Migrating Actor ${actor.name}`);
      await actor.update({"data.untrainedSkill": untrainedSkill._id});
    }
} catch (err) {
  console.error(err);
}

```
and then there is a migration handler that checks which version you are on and makes sure to run those migrations

Not saying that this is "the best way" but just thought that it would be having  less risk to go wrong.. but i might definitely be wrong about that :)